src/unix/notbsd/linux/musl/mod.rs

```diff
@@ -1,67 +1,34 @@
-pub type c_long = i64;
-pub type c_ulong = u64;
-pub type clock_t = i64;
-pub type time_t = i64;
-pub type suseconds_t = i64;
+pub type clock_t = c_long;
+pub type time_t = c_long;
+pub type suseconds_t = c_long;
 pub type ino_t = u64;
 pub type off_t = i64;
 pub type blkcnt_t = i64;
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 pub type c_char = i8;
+
+#[cfg(any(target_arch = "arm", target_arch = "asmjs"))]
+pub type c_char = u8;
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
 pub type wchar_t = i32;
-pub type nlink_t = u64;
-pub type blksize_t = i64;
 
-s! {
-    pub struct stat {
-        pub st_dev: ::dev_t,
-        pub st_ino: ::ino_t,
-        pub st_nlink: ::nlink_t,
-        pub st_mode: ::mode_t,
-        pub st_uid: ::uid_t,
-        pub st_gid: ::gid_t,
-        __pad0: ::c_int,
-        pub st_rdev: ::dev_t,
-        pub st_size: ::off_t,
-        pub st_blksize: ::blksize_t,
-        pub st_blocks: ::blkcnt_t,
-        pub st_atime: ::time_t,
-        pub st_atime_nsec: ::c_long,
-        pub st_mtime: ::time_t,
-        pub st_mtime_nsec: ::c_long,
-        pub st_ctime: ::time_t,
-        pub st_ctime_nsec: ::c_long,
-        __unused: [::c_long; 3],
-    }
+#[cfg(any(target_arch = "arm", target_arch = "asmjs"))]
+pub type wchar_t = u32;
 
-    pub struct stat64 {
-        pub st_dev: ::dev_t,
-        pub st_ino: ::ino64_t,
-        pub st_nlink: ::nlink_t,
-        pub st_mode: ::mode_t,
-        pub st_uid: ::uid_t,
-        pub st_gid: ::gid_t,
-        __pad0: ::c_int,
-        pub st_rdev: ::dev_t,
-        pub st_size: ::off_t,
-        pub st_blksize: ::blksize_t,
-        pub st_blocks: ::blkcnt64_t,
-        pub st_atime: ::time_t,
-        pub st_atime_nsec: ::c_long,
-        pub st_mtime: ::time_t,
-        pub st_mtime_nsec: ::c_long,
-        pub st_ctime: ::time_t,
-        pub st_ctime_nsec: ::c_long,
-        __reserved: [::c_long; 3],
-    }
+// This definition is slightly different on aarch64
+#[cfg(not(target_arch = "aarch64"))]
```

Comment:
Do we actually have aarch64 support?

Stylistically I also tend to prefer copying things around instead of using `#[cfg]` and/or `#[cfg(not)]` (even though it's a little wordier), but doesn't matter too much here, it can just be cleaned up later if it becomes a problem.